File: LeetCode/Simulation/JS/Leetcode6-Z字形变换.js
var convert = function(s, numRows) {
    if(numRows == 1) return s;
    const res = []
    for(let i = 0;i < numRows;i++){
        if(i == 0 || i == numRows - 1){
            for(let j = i;j < s.length;j += 2 * numRows - 2) res.push(s[j]);
        }else{
            for(let j = i,k = 2 * numRows - 2 - i;j < s.length || k < s.length;j += 2 * numRows - 2,k += 2 * numRows - 2) {
                if(j < s.length) res.push(s[j]);
                if(k < s.length) res.push(s[k]);
            }
        }
    }
    return res.join('');
};

File: LeetCode/Simulation/Python/Leetcode6-Z字形变换.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1: return s
        res,tot = [],len(s)
        for i in range(numRows):
            if i == 0 or i == numRows - 1:
                for j in range(i,tot,2 * numRows - 2):
                    res.append(s[j])
            else:
                j,k = i,2 * numRows - 2 - i
                while j < tot or k < tot:
                    if j < tot: res.append(s[j])
                    if k < tot: res.append(s[k])
                    j,k = j + 2 * numRows - 2,k + 2 * numRows - 2
        return ''.join(res) 